lib/ansible/modules/network/nxos/nxos_interface.py

```diff
@@ -279,7 +279,7 @@ def get_interface(intf, module):
 
     command = 'show interface {0}'.format(intf)
     try:
-        body = execute_show_command(command, module)[0]
+        body = run_commands(module, [command])[0]
```

Comment:
@trishnaguha Just to add to the comment by @Qalthos.  I played around with a candidate patch for `execute_show_command` (before you removed it here) that fixed `nxos_interface` for both transport `cli` and `nxapi`.

```python
 def execute_show_command(command, module):
     if module.params['transport'] == 'cli':
-        command += ' | json'
-    cmds = [command]
+        if 'show run' not in command:
+            command += ' | json'
+        cmds = [command]
+        body = run_commands(module, cmds)
+    elif module.params['transport'] == 'nxapi':
+        if 'show run' not in command:
+            output = 'json'
+        else:
+            output = 'text'
+        cmds = [{
+          'command': command,
+          'output': output,
+        }]
     body = run_commands(module, cmds)
     return body
 
-
 def flatten_list(command_lists):
     flat_command_list = []
     for command in command_lists:
```

Ideally it would be nice to do something along the lines of this, regardless of transport.
```python
if 'show run' not in command:
    output = 'json'
else:
    output = 'text'
cmd = [{'command': command, 'output': output,}]
body = run_commands(module, [cmd])[0]
```